services/wfm/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java

```diff
@@ -51,4 +52,20 @@ public RerouteService(RepositoryFactory repositoryFactory) {
         log.info("Get inactive flow ids");
         return flowRepository.findDownFlowIds();
     }
+
+    /**
+     * Get all flow ids for a particular isl.
+     *
+     * @param srcSwitchId source switch id.
+     * @param srcPort     source port.
+     * @param dstSwitchId destination switch id.
+     * @param dstPort     destination port.
+     */
+    public Collection<String> getAllFlowIdsForIsl(SwitchId srcSwitchId, int srcPort,
+                                               SwitchId dstSwitchId, int dstPort) {
+        log.info("Get inactive flow ids");
+        return flowRepository.findAllFlowPairsWithSegment(srcSwitchId, srcPort, dstSwitchId, dstPort).stream()
```

Comment:
why not to do search over flow segments, not flows?

Reply:
I just used an existing method to search for a flow ids by ISL. And by code, we are searching flow ids over flow segments.